PublicMapping/districtbuilder
# Warn when attempting to leave page with selection

## Overview

We want to give the user a chance to save active selections before accidentally closing the page.

### Checklist


### Demo

![screenshot_7](https://user-images.githubusercontent.com/6386/92964201-b7af6c00-f441-11ea-9980-387f66b785e1.png)


### Notes

I provided a custom message, but no browser I tested with is able to display it because of the following.

```
Some browsers used to display the returned string in the confirmation dialog, enabling the event handler to display a custom message to the user. However, this is deprecated and no longer supported in most browsers.
```
Source: https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event

## Testing Instructions

- ./scripts/update
- Load a project
- Verify you can close the page without a confirmation dialog when no geounits are selected
- Select some geounits
- Verify you get a confirmation dialog when attempting to close the page
